com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -1,79 +1,166 @@
+using System;
 using System.Collections.Generic;
 using UnityEngine;
 
 namespace Unity.Netcode.Components
 {
     /// <summary>
-    /// A prototype component for syncing animations
+    /// A prototype component for syncing Mecanim Animator state in a server-driven manner
     /// </summary>
     [AddComponentMenu("Netcode/" + nameof(NetworkAnimator))]
     public class NetworkAnimator : NetworkBehaviour
     {
-
-        private class AnimatorSnapshot : INetworkSerializable
+        private class AnimatorSnapshot : INetworkSerializable, IDisposable
         {
-            public Dictionary<int, bool> BoolParameters;
-            public Dictionary<int, float> FloatParameters;
-            public Dictionary<int, int> IntParameters;
-            public HashSet<int> TriggerParameters;
+            public KeyValuePair<int, bool>[] BoolParamArray;
+            public KeyValuePair<int, float>[] FloatParamArray;
+            public KeyValuePair<int, int>[] IntParamArray;
+            public List<int> TriggerParameters;
             public LayerState[] LayerStates;
 
-            public AnimatorSnapshot(Dictionary<int, bool> boolParameters, Dictionary<int, float> floatParameters, Dictionary<int, int> intParameters, HashSet<int> triggerParameters, LayerState[] layerStates)
+            public AnimatorSnapshot(int boolCount, int floatCount, int intCount, int triggerCount, int layerStatesCount)
             {
-                BoolParameters = boolParameters;
-                FloatParameters = floatParameters;
-                IntParameters = intParameters;
-                TriggerParameters = triggerParameters;
-                LayerStates = layerStates;
+                BoolParamArray = new KeyValuePair<int, bool>[boolCount];
+                IntParamArray = new KeyValuePair<int, int>[intCount];
+                FloatParamArray = new KeyValuePair<int, float>[floatCount];
+                TriggerParameters = new List<int>(triggerCount);
+                LayerStates = new LayerState[layerStatesCount];
+                SetBuffersToDefaultValues();
             }
 
             public AnimatorSnapshot()
             {
-                BoolParameters = new Dictionary<int, bool>(0);
-                FloatParameters = new Dictionary<int, float>(0);
-                IntParameters = new Dictionary<int, int>(0);
-                TriggerParameters = new HashSet<int>();
+                BoolParamArray = new KeyValuePair<int, bool>[0];
+                IntParamArray = new KeyValuePair<int, int>[0];
+                FloatParamArray = new KeyValuePair<int, float>[0];
+                TriggerParameters = new List<int>(0);
                 LayerStates = new LayerState[0];
+                SetBuffersToDefaultValues();
+            }
+
+            private void SetBuffersToDefaultValues()
+            {
+                for (int i = 0; i < BoolParamArray.Length; i++)
+                {
+                    BoolParamArray[i] = new KeyValuePair<int, bool>(-1, false);
```

Comment:
Can we get a constant for `-1` with a meaningful name, that way, we'd know just looking at the code what it means?  e.g. `invalid`, `absent`, or something.